controllers/metal3.io/host_state_machine.go

```diff
@@ -561,6 +566,37 @@ func (hsm *hostStateMachine) handleDeprovisioning(info *reconcileInfo) actionRes
 	return actResult
 }
 
+func (hsm *hostStateMachine) handlePoweringOffBeforeDelete(info *reconcileInfo) actionResult {
+	actResult := hsm.Reconciler.actionPowerOffBeforeDeleting(hsm.Provisioner, info)
+	skipToDelete := func() actionResult {
```

Comment:
nit: most of this function is repeating handleDeprovisioning. It would be great to refactor them.